### ISSUE TITLE
🕐 Support for 75-era customisation (modules, settings, etc.) 🕐

### ISSUE THREAD

**Issue by zach2good** [DerpyProjectGroup/topaz#6549]
**TLDR**

⭐ Please use common sense! ⭐

Through the use of modules, settings, and sane design - content that aims to revert FFXI's gameplay and systems to their states as seen during arbitrary "eras" is allowed and welcomed, as long as it doesn't interfere with the project's main goal of emulating modern retail FFXI and is of sufficiently high quality.

Era content and custom content are not the same. Arbitrary "quality of life" changes do not fall under era - they were never a part of the FFXI retail game.

**Introduction**

- Our first priority is **ALWAYS** to be an FFXI Retail Emulator at the "date of the latest or final patch".
- Customisation systems (modules, settings) can be used to allow arbitrary customisation or an approximation of bygone eras _using the modern unmodified client_. They were originally written to enhance the quality of bug reports back upstream and to allow operators to keep their servers easily updated.
- As long as modifications to approximate or emulate era don't interfere with our main goal (see the first point) and are accurate to what they're trying to recreate then they're welcome.
- Reworking C++ systems to push _relevant_ logic up into Lua to allow for runtime safety and customisation is fine, as long as it's done safely, sanely, and correctly to non-performance-critical systems. Our working assumption is that all FFXI-gameplay-related systems will be tinkered on and theorycrafted forever and the underlying core logic might one day become stable.
- The situation we've always been afraid of and have been trying to avoid is the acceptance of some era system (module or otherwise) and being flooded with PRs from era players/developers constantly tweaking it - taking our time away from the main project.
- It's worth noting that **most** era content still exists and is observable in modern retail (with a few notable exceptions). It is quite rare for an era system or mechanic to be entirely removed. It's more likely that additional systems were added on top and just need to be disabled to get the era behaviour again.

**Questions & Thoughts**

- What is era? Is it tied to the last update before each expansion dropped? Is it tied to the monthly patch notes? Is it just 75-era, 99-era, and 119-era? Certain mechanics, changes, and features came in during random patches during certain expansion lifecycles. How should we mark them?
- Since it is harder to obtain proof for long-gone era gameplay/stats/maths etc. should the quality gating be lower for it? I would argue no - it's all pretty well documented even still.
- How do we alleviate the possible issue of "era system gets tweaked forever and interferes with retail emulation" without bringing in CODEOWNERS, mob rule for PRs, outsourcing to another group/org/repo, or using up excess maintainer time?
- As a rule, we've always avoided client modification, encouraging everyone to have an active retail sub and not to distribute client files. There are one or two era things that _need_ client modification (merit menus, I think?). How do we get around this? Ashita addon that we send a diff to and it applies a temporary diff to client DATs?

**Custom Content**

- As with all content submitted to us - once we merge it we have to support it. We're under no obligation to accept (and then support) any arbitrary custom content that gets submitted to us.

Since this is quite a contested topic I've locked the issue right away. If you want to talk about it or offer up ideas please open a discussion.
